packages/augur-core/src/contracts/symbiote/SymbioteHatchery.sol

```diff
@@ -0,0 +1,143 @@
+import 'ROOT/symbiote/IArbiter.sol';
+import 'ROOT/libraries/Initializable.sol';
+import 'ROOT/para/interfaces/IParaUniverse.sol';
+import 'ROOT/para/interfaces/IParaOICash.sol';
+import 'ROOT/libraries/math/SafeMathUint256.sol';
+
+
+contract SymbioteHatchery is Initializable {
+    using SafeMathUint256 for uint256;
+
+    uint256 private constant MIN_OUTCOMES = 2; // Does not Include Invalid
+    uint256 private constant MAX_OUTCOMES = 7; // Does not Include Invalid
+    uint256 private constant MAX_FEE = 2 * 10**16; // 2%
+    address private constant NULL_ADDRESS = address(0);
+    uint256 private constant MAX_UINT = 2**256 - 1;
+
+    struct Symbiote {
+        address creator;
+        uint256 creatorFee;
+        uint256 numTicks;
+        IArbiter arbiter;
+        ISymbioteShareToken[] shareTokens;
+        uint256 creatorFees;
+    }
+
+    Symbiote[] public symbiotes;
+    ISymbioteShareTokenFactory public tokenFactory;
+    IParaUniverse public paraUniverse;
+    IParaOICash public OICash;
+    IFeePot public feePot;
+    IERC20 public underlyingCurrency;
+
+    event SymbioteCreated();
+    event CompleteSetsMinted();
+    event CompleteSetsBurned();
+    event Claim();
+    event AugurResolutionInitiated();
+
+    function initialize(IParaUniverse _universe, ISymbioteShareTokenFactory _tokenFactory) public returns (bool) {
+        endInitialization();
+        paraUniverse = _universe;
+        tokenFactory = _tokenFactory;
+        OICash = IParaOICash(_universe.openInterestCash());
+        feePot = _universe.getFeePot();
+        underlyingCurrency = IERC20(_universe.cash());
+        underlyingCurrency.approve(_universe.augur(), MAX_UINT);
+        underlyingCurrency.approve(address(feePot), MAX_UINT);
+        return true;
+    }
+
+    function createSymbiote(uint256 _creatorFee, string[] memory _outcomeSymbols, bytes32[] memory _outcomeNames, uint256 _numTicks, IArbiter _arbiter, bytes memory _arbiterConfiguration) public returns (uint256) {
+        require(_numTicks.isMultipleOf(2), "SymbioteHatchery.createSymbiote: numTicks must be multiple of 2");
+        require(_numTicks >= _outcomeSymbols.length, "SymbioteHatchery.createSymbiote: numTicks lower than numOutcomes");
+        require(MIN_OUTCOMES <= _outcomeSymbols.length && _outcomeSymbols.length <= MAX_OUTCOMES, "SymbioteHatchery.createSymbiote: Number of outcomes is not acceptable");
+        require(_outcomeSymbols.length == _outcomeNames.length, "SymbioteHatchery.createSymbiote: outcome names and outcome symbols differ in length");
+        require(_creatorFee <= MAX_FEE, "SymbioteHatchery.createSymbiote: market creator fee too high");
+        {
+            symbiotes.push(Symbiote(
+                msg.sender,
+                _creatorFee,
+                _numTicks,
+                _arbiter,
+                tokenFactory.createShareTokens(_outcomeNames, _outcomeSymbols),
+                0
+            ));
+        }
+        uint256 _id = symbiotes.length-1;
```

Comment:
could this be just before the previous block, so you can omit the subtraction?

Reply:
Yup!